okhttp/src/main/java/com/squareup/okhttp/internal/io/RealConnection.java

```diff
@@ -67,7 +67,7 @@
    * The application layer socket. Either an {@link SSLSocket} layered over {@link #rawSocket}, or
    * {@link #rawSocket} itself if this connection does not use SSL.
    */
-  private Socket socket;
+  public Socket socket;
```

Comment:
I don't have a strong opinion. It's bad practice when widespread, but I don't think it's terrible to do it when you otherwise can't access the internal state that needs verified. You could put a `public Socket getSocket()` on this but that changes nothing in actuality.